packages/payment/src/services/payment-module.ts

```diff
@@ -1,29 +1,283 @@
+    data: CreatePaymentCollectionDTO,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  createPaymentCollection(
+    data: CreatePaymentCollectionDTO[],
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO[]>
+
+  @InjectTransactionManager("baseRepository_")
+  async createPaymentCollection(
+    data: CreatePaymentCollectionDTO | CreatePaymentCollectionDTO[],
+    @MedusaContext() sharedContext?: Context
+  ): Promise<PaymentCollectionDTO | PaymentCollectionDTO[]> {
+    const input = Array.isArray(data) ? data : [data]
+
+    const collections = await this.paymentCollectionService_.create(
+      input,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<PaymentCollectionDTO[]>(
+      Array.isArray(data) ? collections : collections[0],
+      {
+        populate: true,
+      }
+    )
+  }
+
+  updatePaymentCollection(
+    data: UpdatePaymentCollectionDTO[],
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO[]>
+  updatePaymentCollection(
+    data: UpdatePaymentCollectionDTO,
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO>
+
+  @InjectTransactionManager("baseRepository_")
+  async updatePaymentCollection(
+    data: UpdatePaymentCollectionDTO | UpdatePaymentCollectionDTO[],
+    sharedContext?: Context
+  ): Promise<PaymentCollectionDTO | PaymentCollectionDTO[]> {
+    const input = Array.isArray(data) ? data : [data]
+    const result = await this.paymentCollectionService_.update(
+      input,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<PaymentCollectionDTO[]>(
+      Array.isArray(data) ? result : result[0],
+      {
+        populate: true,
+      }
+    )
+  }
+
+  deletePaymentCollection(
+    paymentCollectionId: string[],
+    sharedContext?: Context
+  ): Promise<void>
+  deletePaymentCollection(
+    paymentCollectionId: string,
+    sharedContext?: Context
+  ): Promise<void>
+
+  @InjectTransactionManager("baseRepository_")
+  async deletePaymentCollection(
+    ids: string | string[],
+    @MedusaContext() sharedContext?: Context
+  ) {
+    const paymentCollectionIds = Array.isArray(ids) ? ids : [ids]
+    await this.paymentCollectionService_.delete(
+      paymentCollectionIds,
+      sharedContext
+    )
+  }
+
+  @InjectManager("baseRepository_")
+  async retrievePaymentCollection(
+    paymentCollectionId: string,
+    config: FindConfig<PaymentCollectionDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ): Promise<PaymentCollectionDTO> {
+    const paymentCollection = await this.paymentCollectionService_.retrieve(
+      paymentCollectionId,
+      config,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<PaymentCollectionDTO>(
+      paymentCollection,
+      { populate: true }
+    )
+  }
+
+  @InjectManager("baseRepository_")
+  async listPaymentCollections(
+    filters?: FilterablePaymentCollectionProps,
+    config?: FindConfig<PaymentCollectionDTO>,
```

Comment:
question: Don't we typically default to `{}`
```suggestion
    filters: FilterablePaymentCollectionProps = {},
    config: FindConfig<PaymentCollectionDTO> = {},
```

Reply:
Yes, good catch, fixed!